### PR TITLE
Add program type and attach type to netsh output

### PR DESCRIPTION
**Pull request by dthaler**
* Add program type to netsh "show programs" output
* Add attach type to netsh "show links" output

Signed-off-by: Dave Thaler <dthaler@microsoft.com>

### CODE REVIEW

File: include/ebpf_api.h [hunk 1]
```diff
@@ -491,6 +491,26 @@ extern "C"
...
         _Out_ ebpf_attach_type_t* expected_attach_type);
 
+    /**
+     * @brief Get the name of a given program type.
+     *
+     * @param[in] program_type Program type.
+     *
+     * @returns Name of the program type, or NULL if not found.
+     */
+    _Ret_maybenull_z_ const char*
```
Comment (Alan-Jowett) [inline include/ebpf_api.h:501]: Does the caller need to free the string? Recommend document the contract.

Comment (dthaler) [reply include/ebpf_api.h:501]: the caller does NOT need to free, just like `bpf_object__name()` etc

**Review by Alan-Jowett — APPROVED** [review]

File: libs/api/ebpf_api.cpp [hunk 2]
```diff
@@ -2051,3 +2051,19 @@ ebpf_get_program_type_by_name(
 
     return EBPF_SUCCESS;
 }
+
+_Ret_maybenull_z_ const char*
+ebpf_get_program_type_name(_In_ const ebpf_program_type_t* program_type)
```
Comment (saxena-anurag) [inline libs/api/ebpf_api.cpp:2056]: How will this API work for new program types which are added later? Will we be querying from registry?

**Review by saxena-anurag — APPROVED** [review]

File: libs/api/ebpf_api.cpp (hunk 2 above)
Comment (dthaler) [reply libs/api/ebpf_api.cpp:2056]: Correct, issue #223

**Pull request merged**